assets/js/api.js

```diff
@@ -1,644 +1,368 @@
+  var url;
+  if (neighborhood == "all") {
+    bounds = pdxBounds;
+    neighborhood = '';
+  } else {
+    bounds = '';
   }
-
-  function markerStyle(data) {
-    // console.log(data);
-    var year = parseInt(data.properties['start'].get('year'));
-    return {
-      radius: 8,
-      fillColor: getPermitColor(year),//'#FF5500',
-      color: '#000',
-      weight: 1,
-      opacity: 1,
-      fillOpacity: 0.7
-    };
+  if (dataType == "permits") {
+    url = permitsUrl;
+  } else {
+    url = demolitionsUrl;
   }
-
-  function toggleHoods() {
-    if (hoodsShown) {
-      map.removeLayer(hoods);
-    } else {
-      map.addLayer(hoods);
+  start = typeof start !== 'undefined' ? start : startDate;
+  end = typeof end !== 'undefined' ? end : endDate;
+  //bounds = typeof bounds !== 'undefined' ? bounds : map.getBounds().toBBoxString();
+  type = typeof type !== 'undefined' ? type : "residential";
+  return $.ajax({
+    method: "GET",
+    url: url,
+    data: {
+      type: type, //type: "residential",
+      neighborhood: neighborhood,
+      startdate: start, //moment().subtract(1, 'years').format('YYYY-MM-DD'),
+                        // //"2014-10-04"
+      enddate: end, //moment().format('YYYY-MM-DD') //"2015-10-04"
+      bounds: bounds
     }
-    hoodsShown = !hoodsShown;
-  }
+  })
+  .done(function(data) {
+    var permitsJson = data;
+    var date;
+    // console.log(permitsJson);
+    // permitsLayer.clearLayers();
+    // map.removeControl(timelineLayer.timeSliderControl);
+    // timelineLayer.clearLayers();
+    $(permitsJson.features).each(function(key, data) {
+
+      if (dataType == "permits") {
+        date = data.properties.issuedate;
+      } else {
+        date = data.properties.demolition_date;
+      }
+      var propStartYear = date;//moment(date);//.get('year');
+      var propEndYear = end;//moment(date).add(150, 'days');//.get('year');
+      data.properties['start'] = propStartYear;
+      data.properties['end'] = propEndYear;
 
-  function switchCoords(coordArray) {
-    var temp = coordArray[1];
-    coordArray[1] = coordArray[0];
-    coordArray[0] = temp;
-    return coordArray;
-  }
+    });
+    // permitsLayer.addLayer(permitsLayer[key]);
+    timelineLayer.addData(permitsJson);
+    console.log(this.url);
+    // permitsLayer.addTo(map);
+  })
+  .fail(function() {
+    console.log("Failed to fetch permits json data");
+  });
+  // promises.push(promise);
+}
+
+function getCrimesYear(nbhood, yearRange) {
+  var crimeCount = 0;
+  var crimesByYear = {}; // need year to count mapping to sync count animation to timeline
+  var crimesPromises = [];
+  for (var i = 0; i < yearRange.length; i++) {
+    var promise = $.ajax({
+      method: "GET",
+      url: crimeUrl,
+      data: {
+        query: "perNeighborhoodPerYear",
+        year: yearRange[i],
+        // hardcoding violent crime parameter for the time being:
+        type: "violent"
+      }
+    })
+    .done(function(data) {
```

Comment:
This 'done' callback function is defined inside of for-loop. Can it be extracted outside of the loop and then referenced here?